source/server/well_known_names.h

```diff
@@ -13,6 +13,7 @@ namespace Server {
 class UdpListenerNameValues {
 public:
   const std::string RawUdp = "raw_udp_listener";
+  const std::string Quic = "quic_listener";
```

Comment:
Shouldn't this be defined inside the QUICHE extension code?

Reply:
I thought well_known_names of same category should be defined in one place. No?

Reply:
Yeah, I think this is inline with my comments about where to stick protos.  I mildly prefer having things like this in a well known location, given minimal overhead, but I think we can go either way.

Reply:
I left a comment about this in the other PR, but it seems like we are going through contortions to support QUIC extensions, when I'm doubtful there will ever be another implementation. If we are serious about supporting multiple QUIC extensions, doesn't the name have to be part of the extension, and the factory itself would have to provide an `isQuic()` method or similar?